arches/app/media/js/views/components/workflows/component-based-step.js

```diff
@@ -586,6 +588,9 @@ define([
     function viewModel(params) {
         var self = this;
 
+        this.locked = params.locked;
+        this.setLock = params.setLock;
```

Comment:
Instead of setting/passing down `setLock`, can this be refactored to use a subscription that manipulates `params.setLock`?